### ISSUE TITLE
Mixins break AllocStackTrace

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1580]
The below code causes `str` to have the following AllocStackTrace:

![image](https://user-images.githubusercontent.com/86157825/170907424-ac463e8f-f1a7-435c-8ea3-8439c4499205.png)

Which isn't very helpful at all, and clearly wrong.

Code:
```cs
public static void TestAlloc<T>(T value)
{
	let str = new String();
	Debug.Break();
	delete str;
}

public static mixin TestMixin<T>(T value)
{
	TestAlloc(value);
}

public static mixin OneToTen()
{
	for (int i = 0; i < 10; i++)
		TestMixin!(i);
}

public static mixin TestMixin()
{
	OneToTen!();
}

public static void Main(String[] args)
{
	TestMixin!();
}
```

Tested with https://github.com/beefytech/Beef/commit/087a129007dcaa80933b9a3c308102601c775790

**Comment by bfiete** [comment]
It wasn't exactly trivial to fix, but inlined methods are now handled at f29a74888c6b6dcfa6afa1c7deb4cec08b68d3da

**Comment by disarray2077** [comment]
@bfiete This fix seems to have caused another issue, the following seems be happening consistently in the "Attempted to access deleted object" error:
![image](https://user-images.githubusercontent.com/86157825/171071434-5509be54-51ef-4ff8-8d55-4e7877415b9c.png)

**Comment by bfiete** [comment]
Ah yes- that one is fixed at c076c7d3a6460880e13195a801e4b5a31e4249bc
